### ISSUE TITLE
[Halide:x64-windows] build failure

### ISSUE THREAD

**Issue by alister-chowdhury** [microsoft/vcpkg#13823]
**Host Environment**
- OS: Windows 10 x64
- Compiler: MSVC 2019

**To Reproduce**
`./vcpkg install halide[app,core,mips,nativeclient,opencl,opengl,ptx,rtti,tutorials,utils]:x64-windows`

**Failure logs** 
```
Computing installation plan...
The following packages will be built and installed:
    halide[app,core,mips,nativeclient,opencl,opengl,ptx,rtti,tutorials,utils]:x64-windows
Detecting compiler hash for triplet x64-windows...
Starting package 1/1: halide:x64-windows
Building package halide[app,core,mips,nativeclient,opencl,opengl,ptx,rtti,tutorials,utils]:x64-windows...
Could not locate cached archive: C:\Users\thoth\AppData\Local\vcpkg/archives\c5\c5be98d387f34a28a24e5b89b6f9b179b8d850d5.zip
-- Using cached D:/vcpkg/downloads/halide-Halide-f43293be3725bb959941e38c1b1fa9ae925f7389.tar.gz
-- Cleaning sources at D:/vcpkg/buildtrees/halide/src/ae925f7389-72c34b60dc.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/halide-Halide-f43293be3725bb959941e38c1b1fa9ae925f7389.tar.gz
-- Applying patch fix-install-path.patch
-- Using source at D:/vcpkg/buildtrees/halide/src/ae925f7389-72c34b60dc.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:85 (message):
    Command failed: ninja -v
    Working Directory: D:/vcpkg/buildtrees/halide/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\halide\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:305 (vcpkg_execute_required_process)
  ports/halide/portfile.cmake:50 (vcpkg_configure_cmake)
  scripts/ports.cmake:79 (include)
```

Log file:
```
[1/2] cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/halide/src/ae925f7389-72c34b60dc.clean" "-DWITH_APPS=ON" "-DWITH_TESTS=OFF" "-DWITH_TUTORIALS=ON" "-DWITH_DOCS=OFF" "-DWITH_UTILS=ON" "-DTARGET_NATIVE_CLIENT=ON" "-DTARGET_HEXAGON=OFF" "-DTARGET_METAL=OFF" "-DTARGET_MIPS=ON" "-DTARGET_POWERPC=OFF" "-DTARGET_PTX=ON" "-DTARGET_OPENCL=ON" "-DTARGET_OPENGL=ON" "-DTARGET_OPENGLCOMPUTE=ON" "-DHALIDE_ENABLE_RTTI=ON" "-DTRIPLET_SYSTEM_ARCH=x64" "-DHALIDE_SHARED_LIBRARY=ON" "-DTARGET_X86=ON" "-DTARGET_ARM=OFF" "-DTARGET_AARCH64=OFF" "-DWARNINGS_AS_ERRORS=OFF" "-DCMAKE_MAKE_PROGRAM=D:/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/halide_x64-windows/debug""
FAILED: ../../x64-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/halide/src/ae925f7389-72c34b60dc.clean" "-DWITH_APPS=ON" "-DWITH_TESTS=OFF" "-DWITH_TUTORIALS=ON" "-DWITH_DOCS=OFF" "-DWITH_UTILS=ON" "-DTARGET_NATIVE_CLIENT=ON" "-DTARGET_HEXAGON=OFF" "-DTARGET_METAL=OFF" "-DTARGET_MIPS=ON" "-DTARGET_POWERPC=OFF" "-DTARGET_PTX=ON" "-DTARGET_OPENCL=ON" "-DTARGET_OPENGL=ON" "-DTARGET_OPENGLCOMPUTE=ON" "-DHALIDE_ENABLE_RTTI=ON" "-DTRIPLET_SYSTEM_ARCH=x64" "-DHALIDE_SHARED_LIBRARY=ON" "-DTARGET_X86=ON" "-DTARGET_ARM=OFF" "-DTARGET_AARCH64=OFF" "-DWARNINGS_AS_ERRORS=OFF" "-DCMAKE_MAKE_PROGRAM=D:/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/halide_x64-windows/debug""
-- The C compiler identification is MSVC 19.25.28611.0
-- The CXX compiler identification is MSVC 19.25.28611.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found LLVM 10.0.0
-- Using LLVMConfig.cmake in: D:/vcpkg/installed/x64-windows/share/llvm
Windows: setting CMAKE_OBJECT_PATH_MAX to 260
-- Using llvm-as at D:/vcpkg/installed/x64-windows/bin/llvm-as.exe
-- Using llvm-nm at D:/vcpkg/installed/x64-windows/bin/llvm-nm.exe
-- Using clang at D:/vcpkg/installed/x64-windows/bin/clang.exe
-- Building Halide as a shared library
-- WARNINGS_AS_ERRORS disabled
-- Found OpenMP
-- Found PNG version 
-- Found JPEG version 62
-- Performing Test SUPPORTS_SIZED_DEALLOC
-- Performing Test SUPPORTS_SIZED_DEALLOC - Success
CMake Error at D:/vcpkg/installed/x64-windows/share/llvm/LLVM-Config.cmake:281 (message):
  Library 'Mips' is a direct reference to a target library for an omitted
  target.
Call Stack (most recent call first):
  D:/vcpkg/installed/x64-windows/share/llvm/LLVM-Config.cmake:100 (llvm_map_components_to_libnames)
  D:/vcpkg/installed/x64-windows/share/llvm/LLVM-Config.cmake:93 (explicit_llvm_config)
  src/CMakeLists.txt:773 (llvm_config)


-- Configuring incomplete, errors occurred!
See also "D:/vcpkg/buildtrees/halide/x64-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "D:/vcpkg/buildtrees/halide/x64-windows-dbg/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/halide/src/ae925f7389-72c34b60dc.clean" "-DWITH_APPS=ON" "-DWITH_TESTS=OFF" "-DWITH_TUTORIALS=ON" "-DWITH_DOCS=OFF" "-DWITH_UTILS=ON" "-DTARGET_NATIVE_CLIENT=ON" "-DTARGET_HEXAGON=OFF" "-DTARGET_METAL=OFF" "-DTARGET_MIPS=ON" "-DTARGET_POWERPC=OFF" "-DTARGET_PTX=ON" "-DTARGET_OPENCL=ON" "-DTARGET_OPENGL=ON" "-DTARGET_OPENGLCOMPUTE=ON" "-DHALIDE_ENABLE_RTTI=ON" "-DTRIPLET_SYSTEM_ARCH=x64" "-DHALIDE_SHARED_LIBRARY=ON" "-DTARGET_X86=ON" "-DTARGET_ARM=OFF" "-DTARGET_AARCH64=OFF" "-DWARNINGS_AS_ERRORS=OFF" "-DCMAKE_MAKE_PROGRAM=D:/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/halide_x64-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/halide/src/ae925f7389-72c34b60dc.clean" "-DWITH_APPS=ON" "-DWITH_TESTS=OFF" "-DWITH_TUTORIALS=ON" "-DWITH_DOCS=OFF" "-DWITH_UTILS=ON" "-DTARGET_NATIVE_CLIENT=ON" "-DTARGET_HEXAGON=OFF" "-DTARGET_METAL=OFF" "-DTARGET_MIPS=ON" "-DTARGET_POWERPC=OFF" "-DTARGET_PTX=ON" "-DTARGET_OPENCL=ON" "-DTARGET_OPENGL=ON" "-DTARGET_OPENGLCOMPUTE=ON" "-DHALIDE_ENABLE_RTTI=ON" "-DTRIPLET_SYSTEM_ARCH=x64" "-DHALIDE_SHARED_LIBRARY=ON" "-DTARGET_X86=ON" "-DTARGET_ARM=OFF" "-DTARGET_AARCH64=OFF" "-DWARNINGS_AS_ERRORS=OFF" "-DCMAKE_MAKE_PROGRAM=D:/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/halide_x64-windows""
-- The C compiler identification is MSVC 19.25.28611.0
-- The CXX compiler identification is MSVC 19.25.28611.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found LLVM 10.0.0
-- Using LLVMConfig.cmake in: D:/vcpkg/installed/x64-windows/share/llvm
Windows: setting CMAKE_OBJECT_PATH_MAX to 260
-- Using llvm-as at D:/vcpkg/installed/x64-windows/bin/llvm-as.exe
-- Using llvm-nm at D:/vcpkg/installed/x64-windows/bin/llvm-nm.exe
-- Using clang at D:/vcpkg/installed/x64-windows/bin/clang.exe
-- Building Halide as a shared library
-- WARNINGS_AS_ERRORS disabled
-- Found OpenMP
-- Found PNG version 
-- Found JPEG version 62
-- Performing Test SUPPORTS_SIZED_DEALLOC
-- Performing Test SUPPORTS_SIZED_DEALLOC - Success
CMake Error at D:/vcpkg/installed/x64-windows/share/llvm/LLVM-Config.cmake:281 (message):
  Library 'Mips' is a direct reference to a target library for an omitted
  target.
Call Stack (most recent call first):
  D:/vcpkg/installed/x64-windows/share/llvm/LLVM-Config.cmake:100 (llvm_map_components_to_libnames)
  D:/vcpkg/installed/x64-windows/share/llvm/LLVM-Config.cmake:93 (explicit_llvm_config)
  src/CMakeLists.txt:773 (llvm_config)


-- Configuring incomplete, errors occurred!
See also "D:/vcpkg/buildtrees/halide/x64-windows-rel/CMakeFiles/CMakeOutput.log".
See also "D:/vcpkg/buildtrees/halide/x64-windows-rel/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
```

It's worth mentioning, even though the error references the mips component, if I omit it, I instead get a very similar error referencing the ptx component (NVPTX), and so on and so forth.

**Comment by alister-chowdhury** [comment]
It appears there isn't a dependency on the llv[target-*] items.
Manually vcpkg installing them fixes this.

There are some other errors that occur, but they seem to be related to ninja reporting a subcommand failed (and not really giving any extra info)

**Comment by alexreinking** [comment]
I have fixed this in PR #13860 which also updates Halide to version 10.
